pkg/networkservice/common/connect/server.go

```diff
@@ -25,22 +25,25 @@ import (
 	"time"
 
 	"github.com/golang/protobuf/ptypes/empty"
-	"github.com/networkservicemesh/api/pkg/api/networkservice"
 	"github.com/pkg/errors"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
 
-	"github.com/networkservicemesh/sdk/pkg/networkservice/chains/client"
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+
 	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
 	"github.com/networkservicemesh/sdk/pkg/tools/clienturlctx"
 	"github.com/networkservicemesh/sdk/pkg/tools/grpcutils"
 	"github.com/networkservicemesh/sdk/pkg/tools/log"
 	"github.com/networkservicemesh/sdk/pkg/tools/multiexecutor"
 )
 
+// ClientFactory is used to created new clients when new connection is created.
+type ClientFactory = func(ctx context.Context, cc grpc.ClientConnInterface) networkservice.NetworkServiceClient
+
 type connectServer struct {
 	ctx               context.Context
-	clientFactory     client.Factory
+	clientFactory     ClientFactory
```

Comment:
Why is this changed?

Reply:
`connect` package depend on `chains`/`client` package (but only takes `client.Factory` definition from it), and now that `NewClient` uses `connect` server, we got a dependency loop.